file-formats/aifi/type/zif_aff_aifi_v1.intf.abap

```diff
@@ -0,0 +1,523 @@
+      namespace                    TYPE c LENGTH 15,
+      " (type /aif/ifname)
+      "! <p class="shorttext">AIF Interface Name</p>
+      "! AIF interface name
+      interface_name               TYPE c LENGTH 10,
+      " (type /aif/ifversion)
+      "! <p class="shorttext">AIF Interface Version</p>
+      "! AIF interface version
+      interface_version            TYPE c LENGTH 5,
+      " (type /aif/if_ddic_struct_raw)
+      "! <p class="shorttext">Data Structure</p>
+      "! Data structure
+      "! $required
+      data_structure               TYPE zif_aff_types_v1=>ty_object_name_30,
+      " (type /aif/if_ddic_struct)
+      "! <p class="shorttext">Mapped Data Structure</p>
+      "! Mapped data structure
+      mapped_data_structure        TYPE zif_aff_types_v1=>ty_object_name_30,
+      " (type /aif/msg_tbl)
+      "! <p class="shorttext">Index Table</p>
+      "! Index table
+      "! $required
+      index_table                  TYPE zif_aff_types_v1=>ty_object_name_30,
+      " (type /aif/ifdisplay)
+      "! <p class="shorttext">Interface Display Name</p>
+      "! Interface display name
+      display_name                 TYPE c LENGTH 120,
+      " (relevant types /AIF/MSG_LIFETIME && /AIF/DATA_OLDER_THAN_E)
+      "! <p class="shorttext">Retention Period</p>
+      "! Retention period
+      "! $default '090'
+      retention_period             TYPE c LENGTH 3,
+      "! <p class="shorttext">Expiration Behavior</p>
+      "! Expiration_behavior
+      expiration_behavior          TYPE ty_expiration_behavior_type,
+      " (type /aif/system_field)
+      "! <p class="shorttext">Sending System</p>
+      "! Path for sending system in data structure
+      sending_system               TYPE string,
+      " (type /aif/ifdirection)
+      "! <p class="shorttext">Direction</p>
+      "! Direction
+      direction                    TYPE ty_direction_type,
+      " /AIF/PRE_PROCESSING
+      "! <p class="shorttext">Preprocessing</p>
+      "! Preprocessing
+      uses_preprocessing           TYPE abap_bool,
+      "! <p class="shorttext">Postprocessing</p>
+      "! Postprocessing
+      uses_postprocesssing         TYPE abap_bool,
+    END OF ty_general_information.
+
+  TYPES:
+    "! <p class="shorttext">Recipient</p>
+    "! Recipient
+    BEGIN OF ty_recipient,
+      " (type /aif/recipient_object_name)
+      "! <p class="shorttext">Recipient</p>
+      "! Recipient
+      "! $required
+      recipient TYPE c LENGTH 40,
+    END OF ty_recipient.
+
+  "! <p class="shorttext">Recipients</p>
+  "! Recipients
+  TYPES ty_recipients TYPE STANDARD TABLE OF ty_recipient WITH DEFAULT KEY.
+
+  TYPES:
+    "! <p class="shorttext">General</p>
+    "! General key field settings
+    BEGIN OF ty_key_field_settings,
+      " (type /aif/key_fieldname, field of index table)
+      "! <p class="shorttext">Key Field Name</p>
+      "! Key field name
+      "! $required
+      key_field_name     TYPE zif_aff_types_v1=>ty_object_name_30,
```

Comment:
Also here: When the structure is already about Key Fields this
```suggestion
      " (type /aif/key_fieldname, field of index table)
      "! <p class="shorttext">Name</p>
      "! Key field name
      "! $required
      name     TYPE zif_aff_types_v1=>ty_object_name_30,
```
could be enough.

Reply:
Ok. Now I understand. You´re saying the context of 'name' is given by the structure and doesn´t have to be given for a single field of it. Adjusted. Commit will follow.